CMakeLists.txt

```diff
@@ -74,10 +72,14 @@ endforeach()
 # =================================================================================================
 # OPTIONS
 
+option(CP2K_BUILD_DBCSR "Build dbcsr before building cp2k" OFF)
 option(CP2K_ENABLE_REGTESTS
        "Enable installation of the binaries for running regtests afterwards"
        OFF)
 option(CMAKE_POSITION_INDEPENDENT_CODE "Enable position independent code" ON)
+option(CP2K_ENABLE_CONSISTENCY_CHECKS
```

Comment:
please review it as I may have forgotten something. I basically have it on all the time when I test cp2k. Never encountered an error so far. CMake will fail if there is inconsistency. 

> Nice progress! Once this PR is merged I'll enable CMake as a required check in the CI. I should have done this last week, sorry.

The solution is not perfect. I could also modify cmake such that it only run this test (but no configuration) when the option is turned on (we could run it as a script). So we do not have to create the docker image if the test fails. Basically I would be following the same idea than python but with cmake (that has all information). Drop a word if you consider the solution viable and I can do the necessary adjustments in the PR.

Reply:
replies to comments got mixed up a bit :p

Reply:
> I could also modify cmake such that it only run this test (but no configuration) when the option is turned on (we could run it as a script).

Yes, that would be great! Then we could add it to [test_python.sh](https://github.com/cp2k/cp2k/blob/master/tools/docker/scripts/test_python.sh). We should rename it then to something more generic like `test_misc.sh`. The important point is that these tests does require a toolchain build.

The reason why I hesitated to make the current CMake test obligatory is that it would drive up our cloud costs because it does require the toolchain.

Reply:
it makes perfect sense. I need to add one more test to this as I realized I do half of the work. It does not change the logic though.